src/components/CardComponent/index.data.jsx

```diff
@@ -0,0 +1,20 @@
+import React, { useState, useEffect } from 'react';
+import CardComponentView from './index.view';
+
+import db from '../../db/dexie';
+
+const CardComponentData = () => {
+
+	const [data, setData] = useState("");
+	const [complete, setComplete] = useState(false);
+
+	useEffect(async () => {
+		const result = await db.card.toArray();
+		setData(result);
+		setComplete(true)
```

Comment:
Is the `complete` variable necessary here? It seems like we're coupling the logic that we have some value in `data` with `complete`, which I believe is not necessary.

For line 17, we can check if the data is loaded or not by checking if it is in its default value or not. I would say it might be better to initialize the `data` state to null and `!data && <CardComponentView mainArray = {data} />` to show/hide those cards until loaded.

Reply:
'''!data && <CardComponentView mainArray = {data} /> 
gave some error, so, had to use this one 
(data!=null) && <CardComponentView mainArray = {data} />